client/container_copy.go

```diff
@@ -84,6 +85,17 @@ func (cli *Client) CopyFromContainer(ctx context.Context, containerID, srcPath s
 	if err != nil {
 		return nil, stat, fmt.Errorf("unable to get resource stat from response: %s", err)
 	}
+	if response.header.Get("Content-Encoding") == "gzip" {
+		gr, err := gzip.NewReader(response.body)
+		if err != nil {
+			return nil, stat, err
+		}
+		defer gr.Close()
+		return struct {
+			io.Reader
+			io.Closer
+		}{gr, response.body}, stat, err
```

Comment:
Interesting!

We have `ioutils.ReadCloserWrapper` from `github.com/docker/docker/pkg/ioutils` which does this as well without the anonymous struct. I think it would  be good to use that.

Is it really safe to `defer gr.Close()` here? Doesn't that close the Reader before it's used? I think you would need to use a closer on `ioutils.ReadCloserWrapper` that closes both Readers.

Reply:
Good catch on the closer. I need to setup some stuff to develop on the server side, so I hadn't really tested this.